### PR TITLE
support specifying display names in UCR multibar charts

### PR DESCRIPTION
**Pull request by czue**
related to http://manage.dimagi.com/default.asp?185637

@NoahCarnahan https://github.com/dimagi/commcare-hq/commit/3adba33dc9d3d157cd527ec2cb753607c4867c30 caused all the expanded columns to change to `columnname-choiceindex` in the response json, which a) broke charts that depended on the name and not the index, and b) made it useless to fix since it would be totally opaque what choice the index referred to.

this makes it so you can specify both a column_id and a display (a backlog feature anyways).

cc @proteusvacuum

### CODE REVIEW

File: corehq/apps/reports_core/static/reports_core/js/charts.js [hunk 1]
```diff
@@ -112,19 +112,20 @@ var charts = (function() {
             // initialize records
             for (i = 0; i < config.y_axis_columns.length; i++) {
                 record = {
-                    key: config.y_axis_columns[i],
+                    key: config.y_axis_columns[i]['display'],
```
Comment (dimagimon) [inline corehq/apps/reports_core/static/reports_core/js/charts.js]: ['display'] is better written in dot notation.

File: corehq/apps/reports_core/static/reports_core/js/charts.js [hunk 2]
```diff
@@ -112,19 +112,20 @@ var charts = (function() {
...
                     values: []
                 };
-                valuesDict[config.y_axis_columns[i]] = record;
+                valuesDict[config.y_axis_columns[i]['column_id']] = record;
```
Comment (dimagimon) [inline corehq/apps/reports_core/static/reports_core/js/charts.js]: ['column_id'] is better written in dot notation.

File: corehq/apps/reports_core/static/reports_core/js/charts.js [hunk 3]
```diff
@@ -112,19 +112,20 @@ var charts = (function() {
...
                 chartData.push(record);
             }
+
             for (i = 0; i < data.length; i++) {
                 current = data[i];
                 for (j = 0; j < config.y_axis_columns.length; j++) {
-                    record = valuesDict[config.y_axis_columns[j]];
+                    record = valuesDict[config.y_axis_columns[j]['column_id']];
```
Comment (dimagimon) [inline corehq/apps/reports_core/static/reports_core/js/charts.js]: ['column_id'] is better written in dot notation.

File: corehq/apps/reports_core/static/reports_core/js/charts.js [hunk 4]
```diff
@@ -112,19 +112,20 @@ var charts = (function() {
...
                     record.values.push({
                         x: current[config.x_axis_column] || '',
-                        y: current[config.y_axis_columns[j]]
+                        y: current[config.y_axis_columns[j]['column_id']]
```
Comment (dimagimon) [inline corehq/apps/reports_core/static/reports_core/js/charts.js]: ['column_id'] is better written in dot notation.

**Comment by czue** [comment]
build fail is real - looking

**Comment by czue** [comment]
build fail no longer real. app_manager and ucr tests pass. :white_check_mark:

**Comment by NoahCarnahan** [comment]
Thanks for fixing.

The `column_id` in the graph config refers to a report column, right? If it's easy to add, we could have the chart display value default to the report column's display value.
(not blocking)

**Comment by czue** [comment]
> The `column_id` in the graph config refers to a report column, right? 

right.

> If it's easy to add, we could have the chart display value default to the report column's display value.

yeah, is a good idea. i don't know how easy it is offhand so not going to prioritize for now

**Pull request merged**